packages/library-base/src/commonMain/kotlin/io/realm/Configuration.kt

```diff
@@ -0,0 +1,264 @@
+        protected var name: String = Realm.DEFAULT_FILE_NAME
+        protected var logLevel: LogLevel = LogLevel.WARN
+        protected var removeSystemLogger: Boolean = false
+        protected var userLoggers: List<RealmLogger> = listOf()
+        protected var maxNumberOfActiveVersions: Long = Long.MAX_VALUE
+        protected var notificationDispatcher: CoroutineDispatcher? = null
+        protected var writeDispatcher: CoroutineDispatcher? = null
+        protected var deleteRealmIfMigrationNeeded: Boolean = false
+        protected var schemaVersion: Long = 0
+        protected var encryptionKey: ByteArray? = null
+
+        /**
+         * Creates the RealmConfiguration based on the builder properties.
+         *
+         * @return the created RealmConfiguration.
+         */
+        fun build(): T {
+            REPLACED_BY_IR()
+        }
+
+        /**
+         * Sets the absolute path of the realm file.
+         *
+         * If not set the realm will be stored at the default app storage location for the platform.
+         */
+        fun path(path: String?): S = apply { this.path = path } as S
+
+        /**
+         * Sets the filename of the realm file.
+         *
+         * If setting the full path of the realm this name is not taken into account.
+         */
+        fun name(name: String) = apply { this.name = name } as S
+
+        /**
+         * Sets the classes of the schema.
+         *
+         * The elements of the set must be direct class literals.
+         *
+         * @param classes the set of classes that the schema consists of.
+         */
+        fun schema(classes: Set<KClass<out RealmObject>>) = apply { this.schema = classes } as S
+
+        /**
+         * Sets the classes of the schema.
+         *
+         * The `classes` arguments must be direct class literals.
+         *
+         * @param classes the classes that the schema consists of.
+         */
+        fun schema(vararg classes: KClass<out RealmObject>) =
+            apply { this.schema = setOf(*classes) } as S
+
+        /**
+         * Sets the maximum number of live versions in the Realm file before an [IllegalStateException] is thrown when
+         * attempting to write more data.
+         *
+         * Realm is capable of concurrently handling many different versions of Realm objects, this can e.g. happen if
+         * a flow is slow to process data from the database while a fast writer is putting data into the Realm.
+         *
+         * Under normal circumstances this is not a problem, but if the number of active versions grow too large, it
+         * will have a negative effect on the file size on disk. Setting this parameters can therefore be used to
+         * prevent uses of Realm that can result in very large file sizes.
+         *
+         * @param number the maximum number of active versions before an exception is thrown.
+         * @see [FAQ](https://realm.io/docs/java/latest/.faq-large-realm-file-size)
```

Comment:
The link is `403` ing

Reply:
Yeah. Seems like the docs don't have subsection anchors at their new location. Tried to phrase sub section details with words instead together with a link to the top level FAQ section.